docs/specs/change-password.md

```diff
@@ -0,0 +1,88 @@
+# Change Password
+
+- [Introduction](#introduction)
+- [Use Cases](#use-cases)
+- [Design](#design)
+  - [Change Password](#change-password-1)
+    - [Error Response](#error-response)
+  - [Force Change Password on Next Login](#force-change-password-on-next-login)
+    - [Error Response](#error-response-1)
+
+## Introduction
+
+This document describes the feature for administrators to change the password of a user through the Portal Admin API.
+
+## Use Cases
+
+- An administrator needs to change the password of an existing user.
+- An administrator needs to force a user to change their password.
+- An administrator needs to reset the password of a user who has forgotten their password.
+
+## Design
+
+### Change Password
+
+`resetPassword` mutation is an existing mutation in the Portal Admin API. It will be extended to support following:
+- Send a notification to user to inform them of the password change.
+- Force the user to change their password on next login.
+- Generate a new password automatically. In this case, the new password will be sent to the user through email.
+
+```graphql
+type Mutation {
+  # other root fields...
+
+  resetPassword(
+    input: ResetPasswordInput!
+  ): ResetPasswordPayload!
+}
+
+input ResetPasswordInput {
+  # other fields...
+
+  # Email will be sent to user with new password
+  sendPassword: Boolean
+  # User will be forced to change password on next login
+  forceChangeOnLogin: Boolean
+  # Generate password and send it to user through email
+  generateAndSendPassword: Boolean
```

Comment:
Can you include the full original ResetPasswordInput and specify it again? It is easier to understand to the full picture.

Do not combine generatePassword with sendPassword.

```graphql
input ResetPasswordInput {
  userID: ID!
  # Optional. It was required previously. If password is not given, then it is generated by Authgear
  password: String
  # Optional. Default is false.
  sendPassword: Boolean
  # Optional. Default is false.
  markPasswordAsExpired: Boolean
}
```
With this design, then we do not need validate on the input, every input has meaning.

```
{
  "userID": "userID"
}
```

This means generate password

```
{
  "userID": "userID",
  "password": "password"
}
```

This means change the password to the given value.

The `sendPassword` and `markPasswordAsExpired` flags are now just optional flags, and composable.